lms/djangoapps/grades/tests/test_tasks.py

```diff
@@ -202,7 +202,7 @@ def test_persistent_grades_not_enabled_on_course(self, default_store):
             self.set_up_course(enable_persistent_grades=False)
             self.assertFalse(PersistentGradesEnabledFlag.feature_enabled(self.course.id))
             with check_mongo_calls(0):
-                with self.assertNumQueries(0):
+                with self.assertNumQueries(1):
```

Comment:
My concern wasn't so much about optimizing the query-count when the feature flag is off - since we're planning to turn on the feature for all courses any-day now.
I questioned the underlying reason for needing to query the `PersistOnFailureTask` table during success paths.  It could potentially be a bottleneck and possibly introduce contention even in happy paths - which a client of `PersistOnFailureTask` wouldn't normally expect.

Perhaps this doesn't need to be a merge blocker, but have you considered the following:
1. Update the `reapply_tasks` management command to pass an additional `options` flag to the `apply_async` call.  It seems this may be possible and will be set as a class variable on the Task object (see below).
2. Update `PersistOnFailureTask.on_success` so it's a no-op unless it finds the additional `options` flag.

Per http://docs.celeryproject.org/en/latest/userguide/tasks.html#list-of-options:
```
Any keyword argument passed to the task decorator will actually be set as an attribute of the resulting task class, and this is a list of the built-in attributes.
```